### PR TITLE
Implement upload submission handling and file upload handling

### PR DESCRIPTION
**Pull request by ragnarob**

### CODE REVIEW

File: src/handlers/upload/newUpload.ts [hunk 1]
```diff
@@ -1,22 +1,136 @@
...
+    uploadBody.artistId,
+    uploadBody.classification,
+    uploadBody.category,
+    uploadBody.state,
+    uploadBody.numberOfPages,
+    userId,
+    userIp,
+  ];
+  return { query, values };
+}
```
Comment (ErikMelton) [inline src/handlers/upload/newUpload.ts:88]: No sanitization! SQLi could be possible

File: src/handlers/upload/newUpload.ts [hunk 2]
```diff
@@ -1,22 +1,136 @@
...
+  const insertQuery = `
+    INSERT INTO artist (Name, E621name, PatreonName, IsPending)
+    VALUES (?, ?, ?, ?)
+  `;
+  const insertValues = [
+    newArtist.artistName,
+    newArtist.e621Name || null,
+    newArtist.patreonName || null,
+    true,
+  ];
```
Comment (ErikMelton) [inline src/handlers/upload/newUpload.ts:100]: No sanitization! SQLi could be possible

File: src/handlers/upload/newUpload.ts [hunk 3]
```diff
@@ -1,22 +1,136 @@
...
+  `;
+  const linkInsertValues = [];
+  for (let i = 0; i < linksWithTypes.length; i++) {
+    linkInsertQuery += `(?, ?, ?)`;
+    linkInsertValues.push(newArtistId, linksWithTypes[i].linkUrl, linksWithTypes[i].linkType);
+    if (i < newArtist.links.length - 1) {
+      linkInsertQuery += ', ';
+    }
+  }
+  await queryDb(linkInsertQuery, linkInsertValues);
```
Comment (ErikMelton) [inline src/handlers/upload/newUpload.ts:127]: No sanitization! SQLi could be possible

**Review by ErikMelton — CHANGES_REQUESTED** [review]

File: src/handlers/upload/newUpload.ts (hunk 1 above)
Comment (toastythetoaster) [reply src/handlers/upload/newUpload.ts:88]: SQLi escaping handled by SQL lib

File: src/handlers/upload/newUpload.ts (hunk 2 above)
Comment (toastythetoaster) [reply src/handlers/upload/newUpload.ts:100]: SQLi escaping handled by SQL lib

File: src/handlers/upload/newUpload.ts (hunk 3 above)
Comment (toastythetoaster) [reply src/handlers/upload/newUpload.ts:127]: SQL escaping handled by SQL lib

**Review by ErikMelton — APPROVED** [review]

File: src/handlers/upload/fileUploadHandler.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { NextFunction, Request, Response } from 'express';
+import { makeApiError } from '../../utils/error-handling';
+import { writeGoogleComicFile } from '../../utils/googleStorage';
+
+export async function handleComicFilesUpload(req: Request, res: Response, next: NextFunction) {
+  let files = req.files;
+  let comicName = req.body.comicName;
+  let uploadId = req.body.uploadId;
+
+  if (!files || !files.length) {
```
Comment (toastythetoaster) [inline src/handlers/upload/fileUploadHandler.ts:10]:
```suggestion
  if (!(files && files instanceof Array && files.length > 0)) {
```

**Pull request merged**